clang/lib/Sema/SemaLambda.cpp

```diff
@@ -1723,7 +1725,11 @@ FieldDecl *Sema::BuildCaptureField(RecordDecl *RD,
     // For SYCL compilations, save user specified names for
     // lambda capture.
     if (getLangOpts().SYCLIsDevice || getLangOpts().SYCLIsHost) {
-      StringRef CaptureName = Var->getName();
+      BindingDecl *BD = nullptr;
+      if (!Var)
+        BD = dyn_cast_or_null<BindingDecl>(Val);
```

Comment:
Can we just ask a name from `Val`? It is a `ValueDecl` which is inherited from a `NamedDecl`.

Reply:
Okay, I did that.   I guess why I didn't do it before was because ValueDecl need not just be VarDecls or BindingDecls - which seems to be what we are expecting here.   But perhaps it doesn't matter.